Sources/AsyncHTTPClient/RequestBag+StateMachine.swift

```diff
@@ -532,8 +534,23 @@ extension RequestBag.StateMachine {
         }
     }
 
+    enum DeadlineExceededAction {
+        case cancelScheduler(HTTPRequestScheduler?)
+        case fail(FailAction)
+    }
+
+    mutating func deadlineExceeded() -> DeadlineExceededAction {
+        switch self.state {
+        case .queued(let queuer):
+            self.state = .deadlineExceededWhileQueued
+            return .cancelScheduler(queuer)
+        default:
+            return .fail(self.fail(HTTPClientError.deadlineExceeded))
```

Comment:
Is it correct that we don't change state here?

Reply:
It doesn't look right: historically we transitioned into a terminal state in `fail`, now we don't.

Reply:
We call down to `self.fail(_:)` which does the state transition for us.
 https://github.com/swift-server/async-http-client/blob/5e575855576050753c764145ed6e0068222fea59/Sources/AsyncHTTPClient/RequestBag%2BStateMachine.swift#L558-L601

Reply:
Ah, ok, can we add a comment to that effect? It was not entirely obvious to me.

Reply:
Done in https://github.com/swift-server/async-http-client/pull/601/commits/c7640da4227db0da213c7d88709b13b4e390e715